### PR TITLE
Adjust DDR in preparation for use of OMR's ddrgen

### PR DESCRIPTION
**Pull request by keithc-ca**
Issue: #378
* remove duplicate definition of INTERNAL_PTR_REG_MASK
* recognize types whose names start with "J9SRP(" or "J9WSRP("
* use reflection to find TDumpReader (only available on z/OS)
* workaround java compile problems for openjdk8

### CODE REVIEW

File: debugtools/DDR_VM/src/com/ibm/j9ddr/StructureTypeManager.java [hunk 1]
```diff
@@ -242,12 +242,12 @@ public int getType(String type)
 		}
 		
 		// SRP?
-		if(type.startsWith("J9SRP")) {
+		if (type.equals("J9SRP") || type.startsWith("J9SRP(")) {
```
Comment (pshipton) [inline debugtools/DDR_VM/src/com/ibm/j9ddr/StructureTypeManager.java]: Please explain the reason for this change and the next.

File: debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/AlgorithmPicker.java [hunk 2]
```diff
@@ -26,8 +26,9 @@
  * @author andhall
  *
  */
-public abstract class AlgorithmPicker<T extends IAlgorithm>
+public abstract class AlgorithmPicker<T extends com.ibm.j9ddr.vm29.j9.IAlgorithm>
```
Comment (pshipton) [inline debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/AlgorithmPicker.java]: Its not obvious why this change is required.

File: debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/stackwalker/JITStackWalker.java [hunk 3]
```diff
@@ -82,6 +82,7 @@
 import static com.ibm.j9ddr.vm29.structure.J9JavaAccessFlags.*;
 import static com.ibm.j9ddr.vm29.structure.J9StackWalkConstants.*;
 import static com.ibm.j9ddr.vm29.structure.J9StackWalkState.*;
+import static com.ibm.j9ddr.vm29.structure.MethodMetaDataConstants.INTERNAL_PTR_REG_MASK;
```
Comment (pshipton) [inline debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/stackwalker/JITStackWalker.java]: Its not obvious why this change is required, the constant doesn't appear to be used in this file.

Comment (pshipton) [reply debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/stackwalker/JITStackWalker.java]: I must have made an error searching, I can see the use of it now.

File: debugtools/DDR_VM/src/com/ibm/j9ddr/StructureTypeManager.java (hunk 1 above)
Comment (keithc-ca) [reply debugtools/DDR_VM/src/com/ibm/j9ddr/StructureTypeManager.java]: We want to match types like `J9SRP` or `J9SRP(UDATA)` but not `J9SRP*` or `J9SRPHashTable`. I didn't encounter any troublesome types that start with `J9WSRP`, but it seemed prudent to change that test similarly.

File: debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/AlgorithmPicker.java (hunk 2 above)
Comment (keithc-ca) [reply debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/AlgorithmPicker.java]: In Java 8 builds, the (interim) compiler seemed to get confused about the unqualified type.

File: debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/stackwalker/JITStackWalker.java (hunk 3 above)
Comment (keithc-ca) [reply debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/stackwalker/JITStackWalker.java]: See line 1222.

File: debugtools/DDR_VM/src/com/ibm/j9ddr/StructureTypeManager.java (hunk 1 above)
Comment (pshipton) [reply debugtools/DDR_VM/src/com/ibm/j9ddr/StructureTypeManager.java]: Seems the explanation should be a comment in the code.

File: debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/AlgorithmPicker.java (hunk 2 above)
Comment (pshipton) [reply debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/AlgorithmPicker.java]: Maybe worth a comment?

File: debugtools/DDR_VM/src/com/ibm/j9ddr/StructureTypeManager.java (hunk 1 above)
Comment (keithc-ca) [reply debugtools/DDR_VM/src/com/ibm/j9ddr/StructureTypeManager.java]: Agreed.

File: debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/AlgorithmPicker.java (hunk 2 above)
Comment (keithc-ca) [reply debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/AlgorithmPicker.java]: Will do.

**Comment by keithc-ca** [comment]
Comments added and squashed.

**Comment by pshipton** [comment]
No point in running PR testing since DDR isn't used yet.
